grease/src/Grease/Macaw/Arch.hs

```diff
@@ -245,6 +246,9 @@ data ArchContext arch = ArchContext
   -- Currently, x86-64 is the only supported architecture that does something
   -- non-trivial here. All other architectures can simply leave the stack
   -- pointer by implementing @'_archOffsetStackPointerPostCall' = 'pure'@.
+  , _archABIParams :: [C.Some (MC.ArchReg arch)]
+  -- ^ The ordered integer and pointer ABI registers for the Architecture. In the future
+  -- this should handle things like stack params, and multiple calling conventions.
```

Comment:
Let's make a numbered TODO that references a GitHub issue describing what remains to be done here.